tests/installation/change_desktop.pm

```diff
@@ -20,33 +20,67 @@ sub change_desktop() {
     my ($self) = @_;
     # ncurses offers a faster way
     if (check_var('VIDEOMODE', 'text')) {
-        send_key 'alt-c';
+
+        wait_screen_change {
+            send_key 'alt-c';
+        };
+
         assert_screen 'inst-overview-options';
-        send_key 'alt-s';
+
+        wait_screen_change {
+            send_key 'alt-s';
+        };
     }
     else {
         send_key_until_needlematch 'packages-section-selected', 'tab', 10;
-        send_key 'ret';
+
+        wait_screen_change {
+            send_key 'ret';
+        };
     }
 
     if (check_screen('dependancy-issue', 10) && get_var("WORKAROUND_DEPS")) {
         while (check_screen 'dependancy-issue', 5) {
             if (check_var('VIDEOMODE', 'text')) {
-                send_key 'alt-s', 3;
+
+                wait_screen_change {
+                    send_key('alt-s', 3);
```

Comment:
Better remove the parameter `3` because it's misleading anyway. It's a bool value calling `wait_idle` internally and we want to get rid of all `wait_idle` calls